lib/vanity/adapters/mock_adapter.rb

```diff
@@ -134,17 +132,27 @@ def ab_not_showing(experiment, identity)
       end
 
       def ab_add_participant(experiment, alternative, identity)
-        @experiments[experiment] ||= {}
-        @experiments[experiment][:alternatives] ||= {}
-        alt = @experiments[experiment][:alternatives][alternative] ||= {}
+        alt = alternative(experiment, alternative)
         alt[:participants] ||= Set.new
         alt[:participants] << identity
       end
 
+      def ab_seen(experiment, identity, alternative)
+        if ab_assigned(experiment, identity) == alternative
+          return alternative
```

Comment:
Hm, that looks right. ActiveRecordAdapter looks similar: https://github.com/Futurelearn/vanity/blob/f221e16ebad0d32f74fa5c2a9ca764bcb9b1de66/lib/vanity/adapters/active_record_adapter.rb#L260.

That's confusing 😒